program.py

```diff
@@ -7,1189 +7,369 @@
-            timeout=WEB_REQUEST_TIMEOUT_SECONDS
-        )
-    else:
-        response = requests.get(
-            url,
-            headers=msg_headers,
-            verify=VERIFY_SSL,
-            timeout=WEB_REQUEST_TIMEOUT_SECONDS,
-            auth=(username, password)
-        )
+        raise Exception(f'OMF message was unsuccessful, {message_type}. {response.status_code}:{response.text}')
 
-    # response code in 200s if the request was successful!
-    if response.status_code >= 200 and response.status_code < 300:
-        response.close()
-        print('Value found.  This is unexpected.  "{0}"'.format(
-            response.status_code))
-        print()
-        opId = response.headers["Operation-Id"]
-        status = response.status_code
-        reason = response.text
-        url = response.url
-        error = f"  {status}:{reason}.  URL {url}  OperationId {opId}"
-        raise Exception(f"Check message was failed. {error}")
-    return response.text
 
+# ************************************************************************
+# REQUIRED: retrieves headers for HTTP request to the specified endpoint
+# ************************************************************************
 
-def checkValue(url):
-    # Sends the request out to the preconfigured endpoint..
 
-    global producerToken, sendingToOCS, username, password
+def get_headers(endpoint, compression='', message_type='', action=''):
+    '''Assemble headers for sending to the endpoint's OMF endpoint'''
 
-    # Assemble headers
-    msg_headers = getHeaders()
+    endpoint_type = endpoint["endpoint-type"]
 
-    # Send the request, and collect the response
-    if sendingToOCS:
-        response = requests.get(
-            url,
-            headers=msg_headers,
-            verify=VERIFY_SSL,
-            timeout=WEB_REQUEST_TIMEOUT_SECONDS
-        )
-    else:
-        response = requests.get(
-            url,
-            headers=msg_headers,
-            verify=VERIFY_SSL,
-            timeout=WEB_REQUEST_TIMEOUT_SECONDS,
-            auth=(username, password)
-        )
+    msg_headers = {
+        'messagetype': message_type,
+        'action': action,
+        'messageformat': 'JSON',
+        'omfversion': omfVersion
+    }
 
-    # response code in 200s if the request was successful!
-    if response.status_code < 200 or response.status_code >= 300:
-        response.close()
-        print('Response from endpoint was bad.  "{0}"'.format(
-            response.status_code))
-        print()
-        opId = response.headers["Operation-Id"]
-        status = response.status_code
-        reason = response.text
-        url = response.url
-        error = f"  {status}:{reason}.  URL {url}  OperationId {opId}"
-        raise Exception(f"OMF message was unsuccessful. {error}")
-    return response.text
+    if(compression == 'gzip'):
+        msg_headers["compression"] = 'gzip'
 
+    # If the endpoint is OCS
+    if endpoint_type == EndpointTypes.OCS.value:
```

Comment:
Similar to above, potential here to use a switch. Also, is it necessary to compare to the `.value` of the enum? It looks awkward but you might be doing this correctly.

Reply:
Without the .value you get a string with the format <key>.<value>